server/services/approverCaseloadService.ts

```diff
@@ -97,38 +119,74 @@ export default class ApproverCaseloadService {
 
     const coms = await this.communityService.getStaffDetailsByUsernameList(comUsernames)
 
-    return caseload.map(offender => {
-      const responsibleCom = coms.find(
-        com =>
-          com.username?.toLowerCase() ===
-          offender.licences
-            .find(l => offender.licences.length === 1 || l.status !== LicenceStatus.ACTIVE)
-            .comUsername?.toLowerCase()
-      )
+    return caseload
+      .map(offender => {
+        const responsibleCom = coms.find(
+          com =>
+            com.username?.toLowerCase() ===
+            offender.licences
+              .find(l => offender.licences.length === 1 || l.status !== LicenceStatus.ACTIVE)
+              .comUsername?.toLowerCase()
+        )
+
+        if (responsibleCom) {
+          return {
+            ...offender,
+            probationPractitioner: {
+              staffCode: responsibleCom.staffCode,
+              name: `${responsibleCom.staff.forenames} ${responsibleCom.staff.surname}`.trim(),
+            },
+          }
+        }
+
+        if (!offender.deliusRecord.staff || offender.deliusRecord.staff.unallocated) {
+          return {
+            ...offender,
+          }
+        }
 
-      if (responsibleCom) {
         return {
           ...offender,
           probationPractitioner: {
-            staffCode: responsibleCom.staffCode,
-            name: `${responsibleCom.staff.forenames} ${responsibleCom.staff.surname}`.trim(),
+            staffCode: offender.deliusRecord.staff.code,
+            name: `${offender.deliusRecord.staff.forenames} ${offender.deliusRecord.staff.surname}`.trim(),
           },
         }
-      }
-
-      if (!offender.deliusRecord.staff || offender.deliusRecord.staff.unallocated) {
+      })
+      .map((c: ManagedCaseForApproval) => {
```

Comment:
performing the same mapping that occurred in the frontend

Reply:
(the caseload building code here is now incredibly complex but will simplify in a follow up)